### PR TITLE
Update referral validation numbers and clearing of the answers

### PR DESCRIPTION
**Pull request by kosiakkatrina**
We were clearing out the answers we didn't want to clear upon routing changes because of the enum update

### CODE REVIEW

File: config/forms/2021_2022.json [hunk 1]
```diff
@@ -3222,37 +3222,37 @@
                   "hint_text": "",
                   "type": "radio",
                   "answer_options": {
-                    "0": {
+                    "1": {
```
Comment (baarkerlounger) [inline config/forms/2021_2022.json:3225]: How come this is changing? These should match the data dictionary right?

Comment (kosiakkatrina) [reply config/forms/2021_2022.json:3225]: Yes, I think that these were pre this PR 0,1,2 .... 10 and now they are matching the dictionary? Are we looking at different dictionary files?

File: app/models/case_log.rb [hunk 2]
```diff
@@ -156,7 +156,7 @@ def is_assured_shorthold_tenancy?
   end
 
   def is_internal_transfer?
-    !!(referral && referral.zero?)
+    !!(referral && referral == 1)
```
Comment (baarkerlounger) [inline app/models/case_log.rb]: This can just be `referral == 1`, the presence check is just there for 0 checks because `nil` doesn't respond to `.zero?`

Comment (kosiakkatrina) [reply app/models/case_log.rb]: Nice! Will change that

**Review by baarkerlounger — APPROVED** [review]

**Pull request merged**